File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter
def solution(k, tangerine):
    counter = Counter(tangerine)
    cnts = sorted(counter.values(), reverse = True)
    
    total = 0
    result = 0
    for cnt in cnts:
        total += cnt
        result += 1
        if total >= k:
            break
    return result

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    const counter = {};
    for(const size of tangerine){
        counter[size] = (counter[size] || 0) + 1;
    }
    
    const cnts = Object.values(counter).sort((a, b) => b - a);
    let total = 0, result = 0;
    for(const cnt of cnts){
        total += cnt;
        result++;
        if(total >= k) break;
    }
    return result;
}